patches/lua-tree.patch

```diff
@@ -590,7 +683,44 @@ index 6c92098cc..c3c3c12fe 100644
  
    --- URL escape and format key and value
    -- values should be already decoded or the `raw` option should be passed to prevent double-encoding
-@@ -736,7 +605,7 @@ end
+@@ -508,7 +368,17 @@ do
+   local floor = math.floor
+   local max = math.max
+ 
+-  local is_array_fast = require "table.isarray"
++  local ok, is_array_fast = pcall(require, "table.isarray")
++  if not ok then
++    is_array_fast = function(t)
++      for k in pairs(t) do
++          if type(k) ~= "number" or floor(k) ~= k then
++            return false
++          end
++      end
++      return true
++    end
++  end
+ 
+   local is_array_strict = function(t)
+     local m, c = 0, 0
+@@ -605,7 +475,16 @@ end
+ 
+ 
+ do
+-  local clone = require "table.clone"
++  local ok, clone = pcall(require, "table.clone")
++  if not ok then
++    clone = function(t)
++      local copy = {}
++      for key, value in pairs(t) do
++        copy[key] = value
++      end
++      return copy
++  end
++  end
```

Comment:
Please document this in the patch description.

Reply:
Will defer to [this comment](https://github.com/Kong/goks/pull/27#discussion_r957855744).

Reply:
The general comment `Reintroducing fallback Lua functions, as the OpenResty tooling is not available for use` could be expanded on a bit instead of linking to the Kong Gateway OSS source code. I count three OpenResty `table` usages that could be added to the bullets under that section; WDYT?